src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -315,31 +353,69 @@ export class Selection<T = any> {
     if (this._selectionType !== SelectionType.Multi || !this._items.displayed) {
       return false;
     }
-    const displayedItems: T[] = this._items.displayed;
-    const nbDisplayed = this._items.displayed.length;
+    /** make sure to exclude the locked items from the list */
+    const displayedItems: T[] = this._items.displayed.filter(item => !(this.lockedRefs.indexOf(item) > -1));
+    const nbDisplayed = displayedItems.length;
     if (nbDisplayed < 1) {
       return false;
     }
     const temp: T[] = displayedItems.filter(item => this.current.indexOf(item) > -1);
     return temp.length === displayedItems.length;
   }
 
+  private canItBeLocked() {
+    // We depend on the trackBy and all so there are part of the requirment of is item could be locked
+    return this._selectionType !== SelectionType.None && this._items.all && this._items.trackBy;
```

Comment:
The reason for the other two `&&` is that in some cases the `all` and `trackBy` are not defined. And there is dependency when you want to check the `ref`. It could be the only case that is reproducible into the tests, I did not found any example into the dev app to fail.

With the latest changes, we could go with `this._selectionType !== SelectionType.None`.